### PR TITLE
Add systemtest for livedata/postprocessing interaction

### PR DESCRIPTION
**Pull request by rosswhitfield**
Ref: [5723: LiveDataServer: Add testing in WebMon test environment](https://ornlrse.clm.ibmcloud.com/ccm/resource/itemName/com.ibm.team.workitem.WorkItem/5723)

This adds the livedata container to docker-compose, webmon assume it has ssl so a self-signed ssl cert is created and used for access to livedata. I set a static IP for the nginx container so that it can be accessed consistently from with the docker network and outside.

A systemtest was added that triggers the new data files request, this causes the autoreduction to run with was updated to publish a "plot" to livedata. The data is then verify that it can be accessed from webmon/outside.

You can manually test the interactions. First `make clean` and `make all` then start up docker-compose, `docker-compose up --build`. After which you can access webmon via http://localhost . You will need to visit livedata once by https://172.16.238.222/ and "accept" the self-signed cert (Proceed unsafe) for this to work.

After visit http://localhost/report/arcs/214583/ and you should see the "Example Plot Data" (which is just text).

2024-06-28-144445_1272x1060_scrot

You can then select "reduction" at the bottom of the page then wait, after a while the "plot data" will update and you should see a new timestamp. This verifies that autoreduction is correctly publishing to livedata.

2024-06-28-144519_1272x1060_scrot

This should work for any ARCS run

# Short description of the changes:

# Long description of the changes:

# Check list for the pull request
- [ ] I have read the [CONTRIBUTING]
- [ ] I have read the [CODE_OF_CONDUCT]
- [ ] I have added tests for my changes
- [ ] I have updated the documentation accordingly

# Check list for the reviewer
- [ ] I have read the [CONTRIBUTING]
- [ ] I have verified the proposed changes
- [ ] best software practices
    + [ ] all internal functions have an underbar, as is python standard
    + [ ] clearly named variables (better to be verbose in variable names)
    + [ ] code comments explaining the intent of code blocks
- [ ] All the tests are passing
- [ ] The documentation is up to date
- [ ] code comments added when explaining intent

# Manual test for the reviewer

# References

### CODE REVIEW

**Comment by jmborr** [comment]
@rosswhitfield @backmari can we move from [the current scheme of incorporating the date to the image tag](https://github.com/neutrons/data_workflow/pull/167/files#diff-551d1fcf87f78cc3bc18a7b332a4dc5d8773a512062df881c5aba28a6f5c48d7R53) to the image[ tagging scheme of the livedata repo](https://github.com/neutrons/live_data_server/blob/next/.github/workflows/publish.yml#L45)?

**Review by jmborr — COMMENTED** [review]
@rosswhitfield @backmari can we move from [the current scheme of incorporating the date to the image tag](https://github.com/neutrons/data_workflow/pull/167/files#diff-551d1fcf87f78cc3bc18a7b332a4dc5d8773a512062df881c5aba28a6f5c48d7R53) to the image[ tagging scheme of the livedata repo](https://github.com/neutrons/live_data_server/blob/next/.github/workflows/publish.yml#L45)?

File: Makefile [hunk 1]
```diff
@@ -121,6 +121,11 @@ SNSdata.tar.gz:  ## install SNS data for testing and limited info display
 	# it needs to be removed when the directory changes
 	tar czf SNSdata.tar.gz -C tests/data/ .
 
+ssl: nginx/nginx.crt nginx/nginx.key ## self-signed ssl certificates for livedata server
+
+nginx/nginx.crt nginx/nginx.key:
+	openssl req -x509 -out nginx/nginx.crt -keyout nginx/nginx.key -newkey rsa:2048 -nodes -sha256 --config nginx/san.cnf
```
Comment (jmborr) [inline Makefile:127]: :tada:

**Comment by jmborr** [comment]
@rosswhitfield Upon composing the images in the local environment, the build process complains of missing set variables. 
```
❯ docker-compose up --build
WARNING: The DJANGO_SETTINGS_MODULE variable is not set. Defaulting to a blank string.
WARNING: The LDAP_SERVER_URI variable is not set. Defaulting to a blank string.
WARNING: The LDAP_DOMAIN_COMPONENT variable is not set. Defaulting to a blank string.
WARNING: The CATALOG_URL variable is not set. Defaulting to a blank string.
WARNING: The CATALOG_ID variable is not set. Defaulting to a blank string.
WARNING: The CATALOG_SECRET variable is not set. Defaulting to a blank string.
```
Perhap that's why the containers are unhealthy:
```
ERROR: for dasmon  Container "2b5bbfc386a4" is unhealthy.
ERROR: for nginx  Container "2b5bbfc386a4" is unhealthy.
ERROR: for amq_pv_gen  Container "2b5bbfc386a4" is unhealthy.
ERROR: for livedata  Container "2b5bbfc386a4" is unhealthy.
ERROR: for workflow  Container "2b5bbfc386a4" is unhealthy.
```
Are you setting these using the secret values stored in the webmon-deploy repo?
Which of the settings under `reporting.reporting_app.settings` did you pick for `DJANGO_SETTINGS_MODULE=`

**Comment by rosswhitfield** [comment]
> @rosswhitfield Upon composing the images in the local environment, the build process complains of missing set variables.
> 
> ```
> ❯ docker-compose up --build
> WARNING: The DJANGO_SETTINGS_MODULE variable is not set. Defaulting to a blank string.
> WARNING: The LDAP_SERVER_URI variable is not set. Defaulting to a blank string.
> WARNING: The LDAP_DOMAIN_COMPONENT variable is not set. Defaulting to a blank string.
> WARNING: The CATALOG_URL variable is not set. Defaulting to a blank string.
> WARNING: The CATALOG_ID variable is not set. Defaulting to a blank string.
> WARNING: The CATALOG_SECRET variable is not set. Defaulting to a blank string.
> ```
> 
> Perhap that's why the containers are unhealthy:
> 
> ```
> ERROR: for dasmon  Container "2b5bbfc386a4" is unhealthy.
> ERROR: for nginx  Container "2b5bbfc386a4" is unhealthy.
> ERROR: for amq_pv_gen  Container "2b5bbfc386a4" is unhealthy.
> ERROR: for livedata  Container "2b5bbfc386a4" is unhealthy.
> ERROR: for workflow  Container "2b5bbfc386a4" is unhealthy.
> ```
> 
> Are you setting these variables using the values from the secret CI variables stored in the webmon-deploy repo? Also, which of the settings under `reporting.reporting_app.settings` did you pick for `DJANGO_SETTINGS_MODULE`

Sorry, it looks like the [dev docs](https://data-workflow.readthedocs.io/en/latest/developer/instruction/build.html#running-system-test) need updating.

Do this first before running docker and the tests
```
export DJANGO_SETTINGS_MODULE=reporting.reporting_app.settings.envtest
export LDAP_SERVER_URI=.
export LDAP_DOMAIN_COMPONENT=.
```

You don't need to set the `CATALOG` ones.

**Review by jmborr — APPROVED** [review]
Manual test works. Upon clicking in reduction, a "plot" is uploaded to livedata and the timestamp is updated
capture

**Pull request merged**